lib/wm_util.ml

```diff
@@ -79,12 +79,12 @@ module ETag = struct
     Printf.sprintf "%s" etag
 
   let unescape s =
-    Scanf.sscanf ("\"" ^ s ^ "\"") "%S" (fun u -> u)
```

Comment:
`sscanf` here was used to unescape characters int the string. So if you had a string like `"This has \\\" special characters"`, it will turn it into `"This has \" special characters"`. So I think that should be left alone.

Reply:
I think that the problem may be adding the additional quotes to the string. That might be unnecessary. It's probably worthwhile to add a test. I think porting [this one](https://github.com/inhabitedtype/ocaml-webmachine/blob/master/lib_test/test_logic.ml#L968-L979) would help to resolve this issue. Could you take a crack at that?

Reply:
@seliopou hmm, apart from the `quoted-pair` issue mentioned above, the current behaviour of webmachine is wrong (as expressed in my initial comment) -- a client sending `If-None-Match: "foo"` and an etag function producing the string `foo` won't match. I'm still unsure whether `quoted-pair` is actually used in the wild and how to best support it if needed.

I can look at the test case you mention, and add it to this PR (hopefully later today, atm busy in a meeting).